### PR TITLE
SCCPPGHA-10 Rebrand and forward to the C and C++ action for SQS

### PR DESCRIPTION
**Pull request by antonioaversa**
[SCCPPGHA-10](https://sonarsource.atlassian.net/browse/SCCPPGHA-10)

It rebrands the action, which now forwards to the C and C++ GHA for SQS.
It does not deprecate it yet though: we plan to deprecate both C and C++ GHA for SQS (`sonarqube-github-c-cpp`) and SQC (`sonarcloud-github-c-cpp`) at the same time, once the C and C++ GHA for SQS will forward to the general GHA for SQS (`sonarqube-scan-action`).
Otherwise, people would move to C and C++ GHA for SQS Github(`sonarqube-github-c-cpp`), to just receive another deprecation warning a week or two later :slightly_smiling_face:

Successful runs (against [5fdf7f7](https://github.com/antonioaversa/sonarqube-github-c-cpp/commit/5fdf7f78af70dea876f5fa0c2524364b87e05525) = current master~1):
- `macos-cmake-compdb-gh-actions-sc`: https://github.com/antonioaversa/macos-cmake-compdb-gh-actions-sc/actions/runs/12199217499/job/34032609800
- `windows-msbuild-opencppcoverage-gh-actions-sc`: https://github.com/antonioaversa/windows-msbuild-opencppcoverage-gh-actions-sc/actions/runs/12199521211/job/34033592416
- `linux-cmake-compdb-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-compdb-gh-actions-sc/actions/runs/12199570892/job/34033744740
- `windows-msbuild-vscoverage-gh-actions-sc`: https://github.com/antonioaversa/windows-msbuild-vscoverage-gh-actions-sc/actions/runs/12199575019/job/34033757482
- `macos-xcode-coverage-gh-actions-sc`: https://github.com/antonioaversa/macos-xcode-coverage-gh-actions-sc/actions/runs/12199627229/job/34033919530
- `linux-cmake-gcovr-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-gcovr-gh-actions-sc/actions/runs/12199646317/job/34033979234
- `linux-cmake-llvm-cov-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-llvm-cov-gh-actions-sc/actions/runs/12199683967/job/34034097039
- `macos-xcode-gh-actions-sc`: https://github.com/antonioaversa/macos-xcode-gh-actions-sc/actions/runs/12199696552/job/34034138048
- `macos-cmake-gh-actions-sc`: https://github.com/antonioaversa/macos-cmake-gh-actions-sc/actions/runs/12199709467/job/34034178827
- `windows-msbuild-gh-actions-sc`: https://github.com/antonioaversa/windows-msbuild-gh-actions-sc/actions/runs/12199729189/job/34034241978
- `windows-cmake-gh-actions-sc`: https://github.com/antonioaversa/windows-cmake-gh-actions-sc/actions/runs/12199740389/job/34034277963
- `linux-autotools-gh-actions-sc`: https://github.com/antonioaversa/linux-autotools-gh-actions-sc/actions/runs/12197570550/job/34027493908
- `linux-cmake-gh-actions-sc`: https://github.com/antonioaversa/linux-cmake-gh-actions-sc/actions/runs/12197559985/job/34027461015

List of checkpoint to verify/do before merging a PR.

### Dev checklist

* [x] Write/update tests
* [x] Functional validation
* [x] Public documentation update
* [x] Private documentation update
* [x] Clean commits (should start with a ticket number, clear message, no fixup, wip)
* [x] Maximize code coverage as much as possible

---

### Reviewer checklist

* [ ] Code review
* [ ] Functional validation
* [ ] Check commits are clean

---

### Steps to validate this PR

Fill-in how you performed the validation, and how the reviewer can replicate it:
* [ ] 

[SCCPPGHA-10]: https://sonarsource.atlassian.net/browse/SCCPPGHA-10?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by henryju — APPROVED** [review]
LGTM

File: action.yml [hunk 1]
```diff
@@ -24,102 +26,25 @@ outputs:
...
-        SONAR_SCANNER_SHA_MACOSX_AARCH64: ${{ steps.sonar-scanner-version.outputs.sonar-scanner-sha-macosx-aarch64 }}
-      run: ${GITHUB_ACTION_PATH}/scripts/configure_paths.sh >> $GITHUB_OUTPUT
-
-    - name: Cache sonar-scanner installation
-      id: cache-sonar-tools
-      if: inputs.cache-binaries == 'true'
-      uses: actions/cache@v4
+    - name: SonarQube Cloud Scan
+      id: scan
+      uses: sonarsource/sonarqube-github-c-cpp@25a2efe466506d2293f19f2e3062df4247477464
```
[the remark this answers is not in the document]
Comment (michael-jabbour-sonarsource) [reply action.yml:31]: Thanks for the thorough testing and for highlighting this change. :pray:
Just a quick question for me to improve my understanding: Why is the fix you linked important to have here? My understanding is that it only fixes the test suite URL following a change SQ Next deployment infrastructure. Is the test suite of the SQS action also invoked when running the test suite of the SQC action?

[the remark this answers is not in the document]
Comment (antonioaversa) [reply action.yml:31]: > Why is the fix you linked important to have here?
It's probably not very important to have those changes in - I just wanted to take the latest version of the `sonarqube-github-c-cpp`.
> My understanding is that it only fixes the test suite URL following a change SQ Next deployment infrastructure.
It also changes `actions/cache` from `v3` to `v4`. I am not sure there are relevant changes in `v4`. These are the release notes for `v4`:

> - Integrated with the new cache service (v2) APIs.
> - Updated to node 20

We are going to change this fixed commit to `sonarqube-scan-action@v4.3` with the deprecation in couple of weeks anyway, so I would keep the fixed commit for now, but if you think it's better to point `v2.0.0`, I can do that instead.

[the remark this answers is not in the document]
Comment (michael-jabbour-sonarsource) [reply action.yml:31]: > but if you think it's better to point v2.0.0, I can do that instead.

I think it's good to use the hash, as you suggested. I agree with you about the `actions/cache` upgrade. I just intended this as a question. Sounds good to me!

**Pull request merged**